src/components/progressLinear/progress-linear.scss

```diff
@@ -1,92 +1,110 @@
 $progress-linear-bar-height: 5px !default;
 
-md-progress-linear:not([md-mode="indeterminate"]) {
+md-progress-linear {
   display: block;
+  position: relative;
   width: 100%;
   height: $progress-linear-bar-height;
 
+  padding-top: 0 !important;
+  margin-bottom: 0 !important;
+
   .md-container {
-    overflow: hidden;
+    display:block;
     position: relative;
-    height: $progress-linear-bar-height;
-    top: $progress-linear-bar-height;
-    transform: translate(0, 5px) scale(1, 0);
-    transition: all .3s linear;
-  }
-
-  .md-container.md-ready {
-    transform: translate(0, 0) scale(1, 1);
-  }
+    overflow: hidden;
 
-  .md-bar {
+    width:100%;
     height: $progress-linear-bar-height;
-    position: absolute;
-    width: 100%;
-  }
 
-  .md-bar1, .md-bar2 {
-    transition: all 0.2s linear;
-  }
+    transform: translate(0, 0) scale(1, 1);;
 
-  &[md-mode=determinate] {
-    .md-bar1 {
-      display: none;
-    }
-  }
+    .md-bar {
+      position: absolute;
 
-  &[md-mode=buffer] {
-    .md-container {
-      background-color: transparent !important;
+      left: 0;
+      top: 0;
+      bottom: 0;
+
+      width: 100%;
+      height: $progress-linear-bar-height;
     }
 
     .md-dashed:before {
       content: "";
-      display: block;
+      display: none;
+      position: absolute;
+
+      margin-top: 0;
       height: $progress-linear-bar-height;
       width: 100%;
-      margin-top: 0;
-      position: absolute;
+
       background-color: transparent;
       background-size: 10px 10px !important;
       background-position: 0px -23px;
-      animation: buffer 3s infinite linear;
     }
-  }
 
-  &[md-mode=query] {
-    .md-bar2 {
-      animation: query .8s infinite cubic-bezier(0.390, 0.575, 0.565, 1.000);
+    .md-bar1, .md-bar2 {
+
+      // Just set the transition information here.
+      // Note: the actual transform values are calculated in JS
+
+      transition: transform 0.2s linear;
```

Comment:
I believe using `transform` here instead of `all` disabled the spiffy height transition that makes the component "scale" into view instead of just immediately appearing.